osquery/tables/system/linux/apt_sources.cpp

```diff
@@ -198,6 +210,140 @@ static void genAptSource(const std::string& source,
+    } else {
+      line = input_line;
+    }
+
+    auto colon_pos = line.find(":");
+    if (colon_pos == std::string::npos) {
+      continue;
+    }
+
+    auto key = line.substr(0, colon_pos);
+    auto value = line.substr(colon_pos + 1, std::string::npos);
+    boost::trim(value);
+
+    if (value.empty()) {
+      continue;
+    }
+
+    if (key == "Types") {
+      auto types = osquery::split(value);
+      if (std::find(types.begin(), types.end(), "deb") == types.end()) {
+        return Status::failure("not a deb type repo");
+      }
+    }
+
+    if (key == "Enabled" && value != "on") {
+      return Status::failure("repo is not enabled");
+    }
+
+    if (key == "URIs") {
+      for (auto& uri : osquery::split(value)) {
+        if (uri.find("://") == std::string::npos) {
+          continue;
+        }
+        // Cannot have trailing slashes
+        while (uri.back() == '/') {
+          uri.pop_back();
+        }
+        uris.push_back(uri);
+      }
+    }
+
+    if (key == "Suites") {
+      for (const auto& suite : osquery::split(value)) {
+        suites.push_back(suite);
+      }
+    }
+  }
+
+  if (uris.size() == 0) {
+    return Status::failure("missing valid URIs");
+  }
+
+  if (suites.size() == 0) {
+    return Status::failure("missing Suites");
+  }
+
+  for (const auto& uri : uris) {
+    // All URLs should have protocol, checked earlier
+    auto proto_end = uri.find("://");
+    auto name = uri.substr(proto_end + 3);
+    for (const auto& suite : suites) {
+      auto suite_parts = osquery::split(suite, "/");
+      AptSource apt_source;
+
+      apt_source.base_uri = uri;
+      apt_source.name = name + ' ' + suite;
+      apt_source.cache_file.push_back(name);
+
+      if (suite_parts.size() == 1) {
+        apt_source.cache_file.push_back("dists");
+        apt_source.cache_file.push_back(suite);
+      } else {
+        apt_source.cache_file.insert(apt_source.cache_file.end(),
+                                     suite_parts.begin(),
+                                     suite_parts.end());
+      }
+
+      apt_sources.push_back(apt_source);
+    }
+  }
+
+  return Status::success();
+}
+
+void genDeb822Block(const std::string& source,
+                    const std::string& block,
+                    QueryData& results,
+                    Logger& logger) {
+  std::vector<AptSource> apt_sources;
+  parseDeb822Block(block, apt_sources);
+
+  for (const auto& apt_source : apt_sources) {
+    aptSourceToRow(source, apt_source, results, logger);
+  }
+}
+
+static void genDeb822Source(const std::string& source,
+                            QueryData& results,
+                            Logger& logger) {
+  std::string content;
+
+  auto s = readFile(source, content);
+  if (!s.ok()) {
+    logger.log(google::GLOG_WARNING, s.getMessage());
+    return;
+  }
+
+  std::vector<std::string> blocks;
+  boost::algorithm::split_regex(blocks, content, boost::regex("\n\n"));
```

Comment:
Is the `\n\n` to support multiple stanzas/blocks on the same file? (If so, we should add a test.)

Reply:
Yes it is, correct. I'll add a test